src/endpoint/devices.ts

```diff
@@ -173,6 +210,204 @@ export interface MatterDeviceDetails {
+}
+
+/**
+ * Possible Values for Hub Characteristic "Availability" flags.
+ */
+export type AvailabilityCharacteristic = 'Available' | 'Unavailable' | 'Unsupported'
+
+export type HubData = {
+	/**
+	 * Whether zwaveS2 is enabled for this hub.
+	 */
+	zwaveS2: boolean
+
+	/**
+	 * Whether the hub supports zigbee3.
+	 */
+	zigbee3: boolean
+
+	hardwareType: string
+
+	/**
+	 * Whether the hub allows zigbee unsecure rejoin.
+	 */
+	zigbeeUnsecureRejoin: boolean
+
+	zwaveStaticDsk?: string
+	hardwareId?: string
+
+	/**
+	 * The Zigbee firmware version for this hub.
+	 */
+	zigbeeFirmware: string
+
+	/**
+	 * Describes state of Zigbee OTA.
+	 */
+	zigbeeOta?: string
+
+	/**
+	 * Indicates if Zigbee OTA is enabled.
+	 */
+	otaEnable?: string
+
+	primarySupportAvailability?: AvailabilityCharacteristic
+	secondarySupportAvailability?: AvailabilityCharacteristic
+	zigbeeAvailability?: AvailabilityCharacteristic
+	zwaveAvailability?: AvailabilityCharacteristic
+	threadAvailability?: AvailabilityCharacteristic
+	lanAvailability?: AvailabilityCharacteristic
+	matterAvailability?: AvailabilityCharacteristic
+	localVirtualDeviceAvailability?: AvailabilityCharacteristic
+
+	/**
+	 * The primary hub device id for this hub. Null/Empty if hub is not a repeater.
+	 */
+	primaryHubDeviceId?: string
+
+	zigbeeChannel?: string
+
+	/**
+	 * The Personal Area Network (PAN) id of the zigbee network the hub is on.
+	 */
+	zigbeePanId?: string
+
+	/**
+	 * The unique identifier for this manufactured hub.
+	 */
+	zigbeeEui?: string
+
+	/**
+	 * The short address of a node in Zigbee network.
+	 */
+	zigbeeNodeID?: string
+
+	/**
+	 * The address of a single node in the network.
+	 */
+	zwaveNodeID?: string
+
+	/**
+	 * The common identification of all nodes belonging to one logical Z-Wave network.
+	 */
+	zwaveHomeID?: string
+
+	zwaveSucID?: string
+	zwaveVersion?: string
+
+	/**
+	 * The global region this zwave radio is operating in.
+	 */
+	zwaveRegion?: string
+
+	/**
+	 * MAC address of the hubs network interface.
+	 */
+	macAddress?: string
+
+	/**
+	 * IP address of the hub.
+	 */
+	localIP?: string
+
+	/**
+	 * Whether the hub is zigbee radio functional.
+	 */
+	zigbeeRadioFunctional?: boolean
+
+	/**
+	 * Whether the hub is zwave radio functional.
+	 */
+	zwaveRadioFunctional?: boolean
+}
+
+export type HubDriver = {
+	/**
+	 * The id of the edge driver.
+	 */
+	driverId: string
+
+	/**
+	 * The version of the edge driver.
+	 */
+	driverVersion?: string
+
+	/**
+	 * The id of the edge channel.
+	 */
+	channelId?: string
+}
+
+export type HubDeviceDetails = {
+	/**
+	 * The unique identifier for this manufactured hub.
+	 *
+	 * <^[0-9A-F]{16}$>
+	 */
+	hubEui: string
+
+	/**
+	 * The hub's firmware version.
+	 *
+	 * <^[0-9]\.[0-9]\.[0-9]{3}$>
+	 */
+	firmwareVersion?: string
+
+	hubData: HubData
+
+	/**
+	 * A list of drivers installed on the hub.
+	 */
+	hubDrivers: HubDriver[]
+}
+
+export type EdgeChildDeviceDetails = {
+	driverId: string
```

Comment:
The API documentation doesn't say that the `driverId` is required for an `EdgeChildDeviceDetails` object, although it would make sense to me that it would.

Reply:
Good catch. Fixed.